src/hlstrafe.cpp

```diff
@@ -498,6 +498,29 @@ namespace HLStrafe
 		avec[1] = wishvel[1];
 	}
 
+	void UpdateLookAtViewangle(const PlayerData& player, CurrentState& curState)
+	{
+		curState.TargetYawLookAtOrigin[0] += curState.Parameters.Parameters.LookAt.X;
+		curState.TargetYawLookAtOrigin[1] += curState.Parameters.Parameters.LookAt.Y;
+		curState.TargetYawLookAtOrigin[2] += curState.Parameters.Parameters.LookAt.Z;
+
+		float difference[3];
+		VecSubtract<float, float, 3>(curState.TargetYawLookAtOrigin, player.Origin, difference);
+
+		double yaw = 0;
+		double pitch = 0;
+		if (difference[0] && difference[1])
+			yaw = Atan2(difference[1], difference[0]) * M_RAD2DEG;
+
+		double z = player.Origin[2] - curState.TargetYawLookAtOrigin[2];
+		double xy = Distance<float, float, 2>(curState.TargetYawLookAtOrigin, player.Origin);
+		if (z && xy)
+			pitch = Atan2(z, xy) * M_RAD2DEG;
```

Comment:
When xy is 0 and z is not 0, it still makes sense to set pitch to either +90 or -90. I believe atan2 might already have this behavior, check the docs.